soccer/gameplay/standard_play.py

```diff
@@ -30,13 +39,49 @@ def use_standard_defense(self):
             if self.has_subbehavior_with_name('defense'):
                 self.remove_subbehavior('defense')
 
-    #Handles activity while the play is active. A play wishing to utilize this
-    #method in additionto having an "execute_running" method of its own must call
-    #it via super
+    ## 
+    # Handles activity while the play is active. A play wishing to utilize this
+    # method in addition to having an "execute_running" method of its own must call
+    # it via super
     def execute_running(self):
         self.use_standard_defense()
 
     #Since the standard_play handles defense, it will always handle the goalie
     @classmethod
     def handles_goalie(cls):
         return True
+
+    ##
+    # Call to attempt to preempt the play
+    # Returns true if the preempt is successful 
+    # Override if you want more complex responce to being preempted
+    def try_preempt(self):
+        self.terminate()
+        return True
+
+    ##
+    # Returns true if this play is valid for the passed situation
+    #
+    @classmethod
+    def is_valid(cls, situation):
+        return situation in cls._situationList
+
+    ##
+    # The score function for standard play will check if situation analysis
+    # is enabled, will return a base score based on situation match if it is
+    # or return float('inf') if it is not
+    #
+    # The expected interaction is to override this function can call super()
+    # to get a baseline score, then to modify that score based on situational
+    # factors. If you get a float("inf") back you know that situation analysis
+    # is not running and you will need to overwrite that with some other score
+    #
+    @classmethod
+    def score(cls):
+        if (not main.situationAnalysis.enabled):
+            return float('inf')
```

Comment:
If you want to simplify the score for the time being, it might be easier to just return `0` here, and then in the plays, always return score + whatever was there before. This way you don't have that super ugly if statement in the score function of plays. If you want to keep it separated due to clarity about what's being executed when situationAnalysis is off or on thats very reasonable too

Reply:
I'm still thinking about how to make the score functions cleaner, probably by making some changes in how play selection works and changing the standard_play score function. Right now I think this is the safest way to go since if it is an no-run by default.